src/planner/ConfigurationToEndEffectorOffset.cpp

```diff
@@ -0,0 +1,69 @@
+#include "aikido/planner/ConfigurationToEndEffectorOffset.hpp"
+
+#include "aikido/constraint/Testable.hpp"
+
+namespace aikido {
+namespace planner {
+
+//==============================================================================
+ConfigurationToEndEffectorOffset::ConfigurationToEndEffectorOffset(
+    statespace::ConstStateSpacePtr stateSpace,
+    dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+    const statespace::StateSpace::State* startState,
+    const Eigen::Vector3d& direction,
+    const double distance,
+    constraint::ConstTestablePtr constraint)
+  : Problem(std::move(stateSpace), std::move(constraint))
+  , mEndEffectorBodyNode(std::move(endEffectorBodyNode))
+  , mStartState(startState)
+  , mDirection(direction.normalized())
+  , mDistance(distance)
+{
+  if (direction.norm() < 1e-6)
+    throw std::invalid_argument("Length of direction shouldn't be zero.");
+
+  if (distance < 0.0)
```

Comment:
I personally prefer to be allowed to pass in a negative distance. We can flip the direction and distance in here, if our underlying planners would be unhappy with that.